zetaclient/chains/evm/observer/outbound.go

```diff
@@ -90,49 +99,74 @@ func (ob *Observer) WatchOutbound() {
 			ticker.UpdateInterval(ob.GetChainParams().OutboundTicker, ob.Logger().Outbound)
 		case <-ob.StopChannel():
 			ob.Logger().Outbound.Info().Msg("WatchOutbound: stopped")
-			return
+			return nil
 		}
 	}
 }
 
 // PostVoteOutbound posts vote to zetacore for the confirmed outbound
 func (ob *Observer) PostVoteOutbound(
+	ctx context.Context,
 	cctxIndex string,
 	receipt *ethtypes.Receipt,
 	transaction *ethtypes.Transaction,
 	receiveValue *big.Int,
 	receiveStatus chains.ReceiveStatus,
 	nonce uint64,
-	cointype coin.CoinType,
+	coinType coin.CoinType,
 	logger zerolog.Logger,
 ) {
 	chainID := ob.Chain().ChainId
-	zetaTxHash, ballot, err := ob.ZetacoreClient().PostVoteOutbound(
+	lf := map[string]any{
+		"outbound.chain_id":         chainID,
+		"outbound.external_tx_hash": receipt.TxHash.String(),
+		"outbound.nonce":            nonce,
+	}
+
+	signerAddress := ob.ZetacoreClient().GetKeys().GetOperatorAddress()
+
+	msg := crosschaintypes.NewMsgVoteOutbound(
+		signerAddress.String(),
 		cctxIndex,
 		receipt.TxHash.Hex(),
 		receipt.BlockNumber.Uint64(),
 		receipt.GasUsed,
-		transaction.GasPrice(),
+		math.NewIntFromBigInt(transaction.GasPrice()),
 		transaction.Gas(),
-		receiveValue,
+		math.NewUintFromBigInt(receiveValue),
 		receiveStatus,
-		ob.Chain(),
+		chainID,
 		nonce,
-		cointype,
+		coinType,
 	)
+
+	const gasLimit = zetacore.PostVoteOutboundGasLimit
+
+	var retryGasLimit uint64
+	if msg.Status == chains.ReceiveStatus_failed {
+		retryGasLimit = zetacore.PostVoteOutboundRevertGasLimit
+	}
+
+	zetaTxHash, ballot, err := ob.ZetacoreClient().PostVoteOutbound(ctx, gasLimit, retryGasLimit, msg)
 	if err != nil {
-		logger.Error().
-			Err(err).
-			Msgf("PostVoteOutbound: error posting vote for chain %d nonce %d outbound %s ", chainID, nonce, receipt.TxHash)
-	} else if zetaTxHash != "" {
-		logger.Info().Msgf("PostVoteOutbound: posted vote for chain %d nonce %d outbound %s vote %s ballot %s", chainID, nonce, receipt.TxHash, zetaTxHash, ballot)
+		logger.Error().Err(err).Fields(lf).Msgf("PostVoteOutbound: error posting vote for chain %d", chainID)
+		return
 	}
+
+	lf["outbound.zeta_tx_hash"] = zetaTxHash
+	lf["outbound.ballot"] = ballot
+
+	logger.Info().Fields(lf).Msgf("PostVoteOutbound: posted vote for chain %d", chainID)
```

Comment:
Can we keep the previous logic which prints logs only when it qualifies `else if zetaTxHash != ""`? The reason is that if we don't check empty `zetaTxHash`, we'll be printing lots of logs.